nixos/modules/services/x11/window-managers/xmonad.nix

```diff
@@ -43,6 +43,7 @@ in {
       haskellPackages = mkOption {
         default = pkgs.haskellPackages;
         defaultText = "pkgs.haskellPackages";
+        type = types.package;
```

Comment:
This does not look correct, it is supposed to be scope of haskell packages, not a single package:

```prompt
nix-repl> lib.types.package.check pkgs.haskellPackages
false
```

Reply:
I think @jtojnar is correct here. This breaks the build for my configuration on `master` (which works fine and does not even change this option from the default).

Reply:
Would be cool to have a type for package sets of this kind! But this type is indeed not correct and breaks eval for ppl, please revert it.

Reply:
sorry about that, this should fix it:
https://github.com/NixOS/nixpkgs/pull/111534/files

Reply:
I'm also hitting this issue. This should actually be a function which takes haskellPackages set and returns list of packages.

Reply:
@turboMaCk doesn't seem possible, because the module needs it for `ghcWithPackages` for example. You might be thinking of the `extraPackages` option instead.

Reply:
@roberth yes sorry I missed what line this is. `extraPackages` is the function I've meant. That one internally uses this attribute.